urbanopt/geojson-modelica-translator
# Microgrid templating

#### Any background context you want to provide?
Adds templatized microgrid functionality to gmt (using mbl v9, fwiw)

#### What does this PR accomplish?
- [x] Create new classes for electrical microgrid components
- [x] Templatize the microgrids, based off sample microgrid sys-param file
- [x] Write working mofiles that respect sys-param values

#### How should this be manually tested?
1. Check out master branch of MBL (soon-to-be v9)
1. `poetry run py.test tests/GMT_Lib/test_gmt_lib.py` to generate models from templates.
    - `test_simulate_cooling_plant` will fail because MBLv9 has breaking changes from MBLv8, which that test depends on.
    - The part of the tests that run the microgrid models have been commented out because the models won't run with MSL 3.2, which is all JModelica can do.
    - We'll run the models in Dymola using the following steps:
3. Download [MSLv4](https://github.com/modelica/ModelicaStandardLibrary/releases)
4. Zip [this fork of MSL](https://github.com/beutlich/ModelicaStandardLibrary/tree/add-Dymola-specific-ModelicaServices) to make paths work in Dymola
5. Open Dymola, install both of those zip files (Tools/Library Management/Install)
6. Load MSL, MBL, and the model generated from the template
7. `Simulation/Simulate`
#### What are the relevant tickets?
Resolves https://github.com/urbanopt/urbanopt-modelica/issues/119
